### PR TITLE
Bug 1485152: Add full active_addons list to ClientsDailyView

### PR DESCRIPTION
**Pull request by sunahsuh**
- Removes deprecated active_experiments columns
- Refactors individual udfs into their own files in a new `com.mozilla.telemetry.utils.udfs` package
- Adds a udf for aggregating lists of structs using a column in the struct as a unique index
- Adds the active_addons column to ClientsDailyView

I considered using collect_set but I realized on days where an addon changed (e.g. a version update) we'd end up with duplicate entries for the addon, and given the initial use case for this column is TAAR I think having one entry per addon_id per day makes more sense.

Apologies for the rearrangement in ClientsDailyView, which makes it harder to see the changes -- I let IntelliJ clean up the code so I could clean up imports :/

cc @crankycoder and @benmiroglio as folks who might want to know when this PR gets merged

### CODE REVIEW

File: src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTest.scala [hunk 1]
```diff
@@ -119,6 +123,21 @@ class ClientsDailyViewTest extends FlatSpec with Matchers with DataFrameSuiteBas
     )
   }
 
+  it must "handle 'active_addons' properly" in {
+    test(
+      List(
+        MainSummaryRow(),
+        MainSummaryRow(active_addons = Some(Seq(Addon("a")))),
+        MainSummaryRow(active_addons = Some(Seq(Addon("a"), Addon("b", true), Addon("c")))),
+        MainSummaryRow(active_addons = Some(Seq(Addon("b", false))))
```
Comment (relud) [inline src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTest.scala]: Are addons nullable within the list in main_summary? e.g. `Some(Seq(None)))`
Is the addon id nullable within the list in main_summary? e.g. `Some(Seq(Addon(null)))`

If so, can we add tests for those cases?

File: src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala [hunk 2]
```diff
@@ -21,9 +23,8 @@ object ClientsDailyViewTestHelpers {
     aborts_content: Option[Int] = None,
     aborts_gmplugin: Option[Int] = None,
     aborts_plugin: Option[Int] = None,
+    active_addons: Option[Seq[Addon]] = None,
```
Comment (relud) [inline src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala]: on line 165 I had to use `Option[scala.collection.Seq[SearchCount]]`. I don't recall why, but it definitely broke tests not to.

Could you remove `scala.collection.` there, or add it here, or comment why they are different?

File: src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala [hunk 3]
```diff
@@ -21,9 +23,8 @@ object ClientsDailyViewTestHelpers {
...
     active_addons_count: Option[Long] = None,
-    active_experiment_branch: Option[String] = None,
-    active_experiment_id: Option[String] = None,
```
Comment (relud) [inline src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala]: `active_experiment_id` and `active_experiment_branch` need to be removed at each of their two later occurrences in this file for tests to pass

Comment (sunahsuh) [reply src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala]: Blargh there was a bad last-minute merge when I collided with your changes from yesterday T_T

File: src/main/scala/com/mozilla/telemetry/views/ClientsDailyView.scala [hunk 4]
```diff
@@ -311,4 +188,143 @@ object ClientsDailyView {
...
+      .toSet -- df.columns)
+      // filter Some(missingColumns) to None unless nonEmpty
+      .filter(_.nonEmpty)
+      // filter expressions using missingColumns out of fieldAggregators
+      .map { missingColumns =>
+      logger.warn(s"JOB $jobName $schemaVersion MISSING INPUT COLUMNS: ${missingColumns.mkString(", ")}")
+      fieldAggregators
+        .filter(unresolvedAttributeNames(_)
+          .forall(!missingColumns.contains(_)))
+    }
```
[the remark this answers is not in the document]
Comment (sunahsuh) [reply src/main/scala/com/mozilla/telemetry/views/ClientsDailyView.scala]: Do you mean just moving that closing brace back up inline?

File: src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTest.scala (hunk 1 above)
Comment (sunahsuh) [reply src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTest.scala]: I just looked at the schema definition again and 1. addon_id is required and 2. containsNull is false in the active_addons array definition

File: src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala (hunk 2 above)
Comment (sunahsuh) [reply src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala]: I just took it out and tests passed.. not sure what's up

Comment (relud) [reply src/test/scala/com/mozilla/telemetry/views/ClientsDailyViewTestHelpers.scala]: wfm :D

**Pull request merged**